pkg/analyzer/analyzers/postman/postman.go

```diff
@@ -1,17 +1,116 @@
+//go:generate generate_permissions permissions.yaml permissions.go postman
 package postman
 
 import (
 	"encoding/json"
 	"fmt"
 	"net/http"
 	"os"
+	"strings"
 
 	"github.com/fatih/color"
 	"github.com/jedib0t/go-pretty/table"
 	"github.com/trufflesecurity/trufflehog/v3/pkg/analyzer/analyzers"
 	"github.com/trufflesecurity/trufflehog/v3/pkg/analyzer/config"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/analyzer/pb/analyzerpb"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/context"
 )
 
+var _ analyzers.Analyzer = (*Analyzer)(nil)
+
+type Analyzer struct {
+	Cfg *config.Config
+}
+
+func (Analyzer) Type() analyzerpb.AnalyzerType { return analyzerpb.AnalyzerType_Postman }
+
+func (a Analyzer) Analyze(_ context.Context, credInfo map[string]string) (*analyzers.AnalyzerResult, error) {
+	key, ok := credInfo["key"]
+	if !ok {
+		return nil, fmt.Errorf("missing key in credInfo")
+	}
+	info, err := AnalyzePermissions(a.Cfg, key)
+	if err != nil {
+		return nil, err
+	}
+	return secretInfoToAnalyzerResult(info), nil
+}
+
+func secretInfoToAnalyzerResult(info *SecretInfo) *analyzers.AnalyzerResult {
+	if info == nil {
+		return nil
+	}
+
+	result := analyzers.AnalyzerResult{
+		AnalyzerType:       analyzerpb.AnalyzerType_Postman,
+		Metadata:           nil,
+		UnboundedResources: []analyzers.Resource{},
+	}
+
+	resource := analyzers.Resource{
+		Name:               info.User.User.FullName,
+		FullyQualifiedName: info.User.User.FullName,
+		Type:               "user",
+		Metadata: map[string]any{
+			"role":        strings.Join(info.User.User.Roles, ","),
+			"username":    info.User.User.Username,
+			"email":       info.User.User.Email,
+			"team_name":   info.User.User.TeamName,
+			"team_domain": info.User.User.TeamDomain,
+		},
+	}
+
+	permissions := bakePermissions(info.User.User.Roles)
+
+	// bind all permissions with resources
+	result.Bindings = analyzers.BindAllPermissions(resource, permissions...)
+
+	for _, workspace := range info.Workspace.Workspaces {
+		result.UnboundedResources = append(result.UnboundedResources, analyzers.Resource{
+			Name:               workspace.Name,
+			FullyQualifiedName: workspace.Name,
+			Type:               "workspace",
+			Metadata: map[string]any{
+				"id":         workspace.ID,
+				"type":       workspace.Type,
+				"visibility": workspace.Visibility,
+			},
+		})
+	}
+
+	return &result
+}
+
+func bakePermissions(roles []string) []analyzers.Permission {
+	permissionMap := map[Permission]struct{}{}
+
+	for _, role := range roles {
+		permissions, ok := rolePermission[role]
+		if !ok {
+			continue
+		}
+		for _, permission := range permissions {
+			permissionMap[permission] = struct{}{}
+		}
+	}
+
+	permissions := make([]analyzers.Permission, len(permissionMap))
+	count := 0
+	for perm := range permissionMap {
+		permStr, err := perm.ToString()
+		if err != nil {
+			continue
+		}
+		permissions[count] = analyzers.Permission{
+			Value:  permStr,
+			Parent: nil,
+		}
+		count++
+	}
```

Comment:
**Nit:** Instead of introducing a `count` variable, it might be better to allocate capacity then `append`, especially if we're not always assigning to the slice:

```go
permissions := make([]analyzers.Permission, 0, len(permissionMap))
for perm := range permissionMap {
    permStr, err := perm.ToString()
    if err != nil {
        continue
    }
    permissions = append(permissions, analyzers.Permission{
        Value: permStr,
    })
}
```

Reply:
Although in this case, `err` *should* always be `nil`. But this pattern is generally less error prone when the loop body is non-trivial imo.